unity-renderer/Assets/DCLServices/EmotesCatalog/EmotesCatalogService.cs

```diff
@@ -236,10 +273,28 @@ public void ForgetEmotes(IList<string> ids)
         }
     }
 
+    public async UniTask<EmbeddedEmotesSO> GetEmbeddedEmotes()
+    {
+        if(embeddedEmotesSO == null)
+            await UniTask.WaitUntil(() => embeddedEmotesSO != null);
+        return embeddedEmotesSO;
+    }
+
     public void Dispose()
     {
         bridge.OnEmotesReceived -= OnEmotesReceived;
         bridge.OnEmoteRejected -= OnEmoteRejected;
         bridge.OnOwnedEmotesReceived -= OnOwnedEmotesReceived;
+        DisposeCT();
+    }
+
+    private void DisposeCT()
+    {
+        if (addressableCTS != null)
+        {
+            addressableCTS.Cancel();
+            addressableCTS.Dispose();
+            addressableCTS = null;
```

Comment:
If this token will be created in the constructor, then you don't need the `null` check. Also, I think it is not necessary to assign null at line 297, because Dispose itself will nullify it (at least I expect such behaviour)